File: Betkibans.Server/Models/Entities/Product.cs
namespace Betkibans.Server.Models.Entities;

public class Product
{
    public int ProductId { get; set; }
    public int SellerId { get; set; }
    public int CategoryId { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public decimal Price { get; set; }
    public int StockQuantity { get; set; }
    public decimal? Length { get; set; }
    public decimal? Width { get; set; }
    public decimal? Height { get; set; }
    public decimal? Weight { get; set; }
    public bool IsActive { get; set; } = true;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? UpdatedAt { get; set; }
    
    // Calculated Properties
    public decimal AverageRating { get; set; } = 0;
    public int TotalReviews { get; set; } = 0;
    
    // Navigation Properties
    public Seller Seller { get; set; } = null!;
    public Category Category { get; set; } = null!;
    public ICollection<ProductImage> ProductImages { get; set; } = new List<ProductImage>();
    public ICollection<ProductMaterial> ProductMaterials { get; set; } = new List<ProductMaterial>();
    public ICollection<CartItem> CartItems { get; set; } = new List<CartItem>();
    public ICollection<OrderItem> OrderItems { get; set; } = new List<OrderItem>();
    public ICollection<Review> Reviews { get; set; } = new List<Review>();
}

File: Betkibans.Client/src/types/Product.ts
export interface ProductImage {
    productImageId: number;
    imageUrl: string;
    isPrimary: boolean;
}

export interface Product {
    productId: number;
    name: string;
    description: string;
    price: number;
    stockQuantity: number;
    categoryName: string;
    categoryId: number;
    sellerBusinessName?: string;
    discountPrice?: number;

    // Images
    productImages: ProductImage[];
    productMaterials?: { materialId: number; materialName: string }[];
    
    length?: number;
    width?: number;
    height?: number;
    weight?: number;
    
    color?: string;
    finishType?: string;
    craftingTimeDays?: string;
    careInstructions?: string;
    careWarnings?: string;

    // Optional fields
    materials?: string[];
    averageRating: number;
    totalReviews: number;
}